src/base/bittorrent/session.cpp

```diff
@@ -4662,15 +4667,19 @@ void Session::handleStateUpdateAlert(const lt::state_update_alert *p)
 
         torrent->handleStateUpdate(status);
         updatedTorrents.push_back(torrent);
+
+        m_globalDownloadRate += torrent->downloadPayloadRate();
+        m_globalUploadRate += torrent->uploadPayloadRate();
```

Comment:
Are you sure you can take into account updated torrents only? Is it impossible that some other torrents have non zero rates?

Reply:
I don't know maybe @arvidn can help here?

Reply:
in principle I believe it's possible for a torrent to have a non-zero transfer rate that doesn't change, and not be included in an update. It's extremely unlikely to happen though, since transfer rates are specified in bytes per second

Reply:
btw, do we understand where the discrepancy is between the session transfer rates and the sum of the torrent rates? does the session not have a counter for payload-only transfers?

Reply:
or is it just the update intervals being different?

Reply:
> do we understand where the discrepancy is between the session transfer rates and the sum of the torrent rates?

I think the problem is that the session stats update and torrent update can't be taken at the same "time point" causing discrepancy like these https://github.com/qbittorrent/qBittorrent/issues/12459 https://github.com/qbittorrent/qBittorrent/issues/8738  (the tab bottom speed is from post_session_stats while torrent speed are updated from post_torrent_updates)

> is it just the update intervals being different

we try to always call post_torrent_updates and post_session_updates together but the problem (https://github.com/qbittorrent/qBittorrent/issues/12459 https://github.com/qbittorrent/qBittorrent/issues/8738) still persists.